src/core/gpu_vk.c

```diff
@@ -3417,7 +3449,17 @@ static gpu_memory* allocate(gpu_memory_type type, VkMemoryRequirements info, VkD
       }
 
       allocator->block = memory;
-      allocator->cursor = info.size;
+
+      // Mark the initial region
+      allocator->pageCount = blockSize / GPU_PAGE_SIZE;
```

Comment:
Note that this can technically be derived from the memory type / allocator index, so it could be removed if needed.  It may still be useful for convenience or for oversized blocks, and `gpu_allocator` is much bigger now so we don't really need to save the 4 bytes.